src/Compilers/CSharp/Portable/Declarations/DeclarationTreeBuilder.cs

```diff
@@ -265,18 +274,64 @@ private static ImmutableArray<ReferenceDirective> GetReferenceDirectives(Compila
             return decl;
         }
 
+        private static QuickAttributes GetQuickAttributes(
+            SyntaxList<UsingDirectiveSyntax> usings, bool global)
+        {
+            var result = QuickAttributes.None;
+
+            foreach (var directive in usings)
+            {
+                if (directive.Alias == null)
+                    continue;
+
+                var isGlobal = directive.GlobalKeyword.Kind() != SyntaxKind.None;
+                if (isGlobal != global)
+                    continue;
+
+                result |= GetQuickAttributes(directive.Name.GetUnqualifiedName().Identifier.ValueText, inAttribute: false);
+            }
+
+            return result;
+        }
+
         public override SingleNamespaceOrTypeDeclaration VisitCompilationUnit(CompilationUnitSyntax compilationUnit)
         {
             if (_syntaxTree.Options.Kind != SourceCodeKind.Regular)
             {
                 return CreateScriptRootDeclaration(compilationUnit);
             }
 
+            _nonGlobalAliasedQuickAttributes = GetNonGlobalAliasedQuickAttributes(compilationUnit);
+
             var children = VisitNamespaceChildren(compilationUnit, compilationUnit.Members, ((Syntax.InternalSyntax.CompilationUnitSyntax)(compilationUnit.Green)).Members);
 
             return CreateRootSingleNamespaceDeclaration(compilationUnit, children, isForScript: false);
         }
 
+        private static QuickAttributes GetNonGlobalAliasedQuickAttributes(CompilationUnitSyntax compilationUnit)
+        {
+            var result = GetQuickAttributes(compilationUnit.Usings, global: false);
+            foreach (var member in compilationUnit.Members)
+            {
+                if (member is BaseNamespaceDeclarationSyntax @namespace)
+                    result |= GetNonGlobalAliasedQuickAttributes(@namespace);
+            }
+
+            return result;
+        }
+
+        private static QuickAttributes GetNonGlobalAliasedQuickAttributes(BaseNamespaceDeclarationSyntax @namespace)
+        {
+            var result = GetQuickAttributes(@namespace.Usings, global: false);
+            foreach (var member in @namespace.Members)
+            {
+                if (member is BaseNamespaceDeclarationSyntax child)
+                    result |= GetNonGlobalAliasedQuickAttributes(child);
```

Comment:
nit: please match surrounding code for braces (here and elswhere in the change)

Reply:
i can't figure out the compiler rules about braces :)

Reply:
made all the braces consistent in the file.

Reply:
I'm unclear how this is helping. Now the PR is full of irrelevant style changes... :-/

Reply:
But they are consistent now :)

Reply:
Between (1) putting braces, or (2) matching the style of the file, which are the conventions as I understand them, you chose (3) change the style of the file (which we generally push back on). Sorry but this is aggravating

Reply:
i've added braces to all constructs.